xbee/tests/test_ieee.py

```diff
@@ -681,16 +682,17 @@ def test_read_empty_string(self):
         an empty string. In this event, we must not crash.
         """
         
-        class BadReadDevice(FakeReadDevice):
+        class BadReadDevice(Serial):
             def __init__(self, bad_read_index, data):
                 self.read_id = 0
                 self.bad_read_index = bad_read_index
-                super(BadReadDevice, self).__init__(data)
+                super(BadReadDevice, self).__init__()
+                self.set_read_data(data)
             
-            def inWaiting(self):
+            def in_waiting(self):
```

Comment:
This is never called. I don't think it should have been changed from `inWaiting`. The test passes either way so I'm not sure why this is even here. I say either delete it or change it back.